### ISSUE TITLE
Varnish config conflicts with Drupal's stage_file_proxy module

### ISSUE THREAD

**Issue by Leksat** [uselagoon/lagoon-images#239]
TLDR: Drupal's Stage File Proxy module + Lagoon's Varnish = circular redirects

Drupal's [Stage File Proxy](https://www.drupal.org/project/stage_file_proxy) module response with 302 redirects when it encounters a file that was not yet downloaded to the current environment. Also, it tries to prevent possible issues with reverse proxies by settings the following header:
```
Cache-Control: must-revalidate, no-cache, post-check=0, pre-check=0, private
```
(See https://github.com/drupalprojects/stage_file_proxy/blob/8.x-1.0-alpha3/src/EventSubscriber/ProxySubscriber.php#L140-L155)

In Lagoon, the `Cache-Control` header in unset for all file requests:
https://github.com/amazeeio/lagoon/blob/7d1d265b41211be5f493b80a53aa158f1d60d7f8/images/varnish-drupal/drupal.vcl#L308-L312

As a result: 302 redirects are cached, and we have circular redirects.

My guess is that Stage File Proxy behaves correctly, but Varnish config needs to be updated to exclude 301 and 302 redirects from the `Don't allow static files to set cookies` rule.
